presto-spi/src/main/java/com/facebook/presto/spi/statistics/ColumnStatistics.java

```diff
@@ -11,38 +11,54 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
 package com.facebook.presto.spi.statistics;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Consumer;
+
 import static com.facebook.presto.spi.statistics.Estimate.unknownValue;
+import static java.util.Collections.singletonList;
+import static java.util.Collections.unmodifiableList;
 import static java.util.Objects.requireNonNull;
 
 public final class ColumnStatistics
 {
-    private final Estimate dataSize;
-    private final Estimate nullsCount;
-    private final Estimate distinctValuesCount;
+    private static final List<RangeColumnStatistics> SINGLE_UNKNOWN_RANGE_STATISTICS = singletonList(RangeColumnStatistics.builder().build());
 
-    private ColumnStatistics(Estimate dataSize, Estimate nullsCount, Estimate distinctValuesCount)
-    {
-        this.dataSize = requireNonNull(dataSize, "dataSize can not be null");
-        this.nullsCount = requireNonNull(nullsCount, "nullsCount can not be null");
-        this.distinctValuesCount = requireNonNull(distinctValuesCount, "distinctValuesCount can not be null");
-    }
+    private final Estimate nullsFraction;
+    private final List<RangeColumnStatistics> rangeColumnStatistics;
 
-    public Estimate getDataSize()
+    private ColumnStatistics(Estimate nullsFraction, List<RangeColumnStatistics> rangeColumnStatistics)
     {
-        return dataSize;
+        this.nullsFraction = requireNonNull(nullsFraction, "nullsFraction can not be null");
+        requireNonNull(rangeColumnStatistics, "rangeColumnStatistics can not be null");
+        if (!rangeColumnStatistics.stream().allMatch(Objects::nonNull)) {
+            throw new NullPointerException("elements of rangeColumnStatistics can not be null");
+        }
+        if (rangeColumnStatistics.size() > 1) {
+            // todo add support for multiple ranges.
+            throw new IllegalArgumentException("Statistics for multiple ranges are not supported");
```

Comment:
That is an option. But IMO we can add that later. This will not change signatures of constructor/builder. Just some calls which result in exception now, will start passing.